python/ray/data/dataset.py

```diff
@@ -1135,15 +1146,15 @@ def max(self, on: Union[KeyFn, List[KeyFn]] = None) -> U:
             - ``on=["col_1", ..., "col_n"]``: an n-column ``ArrowRow``
               containing the column-wise max of the provided columns.
 
-            If the dataset is empty, then a ``ValueError`` is raised.
+            If the dataset is empty, all values are null, or any value is null
+            AND ``ignore_nulls`` is ``False``, then the output will be None.
```

Comment:
This comment doesn't match the previous `ignore_nulls` comment: do we raise an error or return None if any value is null AND `ignore_nulls` is False?

Reply:
We return `None` now, in an attempt to match the semantics of Pandas and NumPy. Let me update the PR description with the semantics detailed in this comment: https://github.com/ray-project/ray/pull/20787#discussion_r760431754

Reply:
Ah I see that I need to update the `ignore_nulls` arg comments in the docstrings! Let me update that.